### PR TITLE
[Middleware] Move to GenericHost

### PR DESCRIPTION
**Pull request by BrennanConroy**
Part of https://github.com/dotnet/aspnetcore/issues/20964

Each Middleware has its own commit

Edit:
Disposing the `IHost` for all tests is in the last commit, sorry did it as an after the fact change so it's not separated by area.

### CODE REVIEW

File: src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs [hunk 1]
```diff
@@ -17,141 +18,177 @@ public class AcceptLanguageHeaderRequestCultureProviderTest
...
+            using (var server = host.GetTestServer())
             {
                 var client = server.CreateClient();
                 client.DefaultRequestHeaders.AcceptLanguage.ParseAdd("jp,ar-SA,en-US");
                 var count = client.DefaultRequestHeaders.AcceptLanguage.Count;
                 var response = await client.GetAsync(string.Empty);
                 Assert.Equal(3, count);
             }
+
+            await host.StopAsync();
```
Comment (Tratcher) [inline src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs:58]: Remind me why you had to call stop? Why isn't Dispose good enough?

**Review by Tratcher — APPROVED** [review]

File: src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs (hunk 1 above)
Comment (BrennanConroy) [reply src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs:58]: I don't remember exactly, I think this was the issue where memory blocks were not being disposed which breaks in debug.

Comment (Tratcher) [reply src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs:58]: That's only an issue in Kestrel and SignalR (?), not TestServer.

Comment (BrennanConroy) [reply src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs:58]: Ah true, only if you use Kestrel. Want me to remove the `StopAsync`'s then?

Comment (Tratcher) [reply src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs:58]: Remove a few just to confirm it works without them.

Comment (BrennanConroy) [reply src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs:58]: Seems to work fine. I'll remove them.

**Pull request merged**